web/_js/main/infoblock.js

```diff
@@ -158,10 +158,78 @@ function createInfoBlock(entry, mode = 0) {
 		idElementContainer.appendChild(editElement)
 	}
 
+	// Adds giscus button
+	if (entry.id >= 0) {
+		const giscusButton = createGiscusButton(entry)
+		idElementContainer.appendChild(giscusButton)
+	}
+
 	// Removes empty elements
 	if (!bodyElement.hasChildNodes()) bodyElement.remove()
 	if (!linkListElement.hasChildNodes()) linkListElement.remove()
 	if (!listElement.hasChildNodes()) listElement.remove()
 
 	return element
+}
+
+function createGiscusButton(entry) {
+	const button = document.createElement("button")
+	button.className = "btn btn-sm btn-outline-primary"
+	button.innerHTML = '<i class="bi bi-chat-dots" aria-hidden="true"></i> Comments'
+	button.addEventListener("click", () => showCommentsModal(entry))
+	return button
+}
+
+function showCommentsModal(entry) {
+	const modal = createModal(entry)
+	document.body.appendChild(modal)
+	
+	const bsModal = new bootstrap.Modal(modal)
+	bsModal.show()
+	
+	modal.addEventListener('shown.bs.modal', () => initGiscus(modal, entry))
+	modal.addEventListener('hidden.bs.modal', () => modal.remove())
+}
+
+function createModal(entry) {
+	const modal = document.createElement("div")
+	modal.className = "modal fade"
+	modal.tabIndex = -1
+	modal.innerHTML = `
+		<div class="modal-dialog modal-lg">
+			<div class="modal-content">
+				<div class="modal-header">
+					<h5 class="modal-title">Comments for ${entry.id}</h5>
```

Comment:
Despite only using the entity ID to associate the comments with Giscus, we should still use `entry.name` to display the name alongside the ID in the modal title.

Reply:
Oh good point